### DISCUSSION TITLE
Can we use Android as a drone controller?

### DISCUSSION THREAD

**Discussion by starqus** [DroidDrone/DroidDrone#1] [Ideas]
Can we use Android as a drone controller? So, I think, there is no need to use a dedicated flight controller with this project. There are sensors on the phone, we can just connect the motor driver via USB. Additionally, some phones can use multiple cameras. We can use low resolution cameras to keep the drone stable. The best project I've seen in a long time. It can turn Android into a hardware-independent robot operating system. I want to be involved in this project.

**Comment by IvanSchulz** [comment]
Hi @maviation, thank you. I also have this idea and it should work in theory, I think. But it is very difficult because we have to write our own FC for Android. In the solution we can take advantage of INAV or ArduPilot (release soon), and it's a bit safer - if something goes wrong with Android or app, we still have an independent FC with RTH function.

**Reply by IvanSchulz** [reply]
And we still need a microcontroller (like a Raspberry Pico or similar) to interface between Android and hardware (ESC, current/voltage sensor etc.)

**Reply by starqus** [reply]
Yes you are right. Android can freeze. Better to be safe. But now I am looking into the DJI system and it looks like the sensors are connected to the image procressor. https://tr.ifixit.com/Guide/DJI+Mini+3+Chip+ID/160247
